components/Modal/CreateImageUpload.js

```diff
@@ -0,0 +1,751 @@
+  }
+
+  componentWillMount() {
+    this.props.fetchingUploadProviders();
+    if (this.props.imageTypes.length === 0) {
+      this.props.fetchingComposeTypes();
+    }
+    if (this.props.composeQueueFetched === false) {
+      this.props.fetchingQueue();
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.clearQueue();
+  }
+
+  setNotifications() {
+    this.props.layout.setNotifications();
+  }
+
+  setUploadService(_, event) {
+    this.setState({ uploadService: event.target.value });
+  }
+
+  setUploadSettings(value, event) {
+    const key = event.target.id;
+    this.setState(prevState => ({ uploadSettings: Object.assign({}, prevState.uploadSettings, { [key]: value }) }));
+  }
+
+  setImageType(imageType) {
+    this.setState({
+      imageType,
+      uploadService: ""
+    });
+  }
+
+  setImageName(imageName) {
+    this.setState({ imageName });
+  }
+
+  isPendingChange() {
+    return (
+      this.props.blueprint.workspacePendingChanges.length > 0 || this.props.blueprint.localPendingChanges.length > 0
+    );
+  }
+
+  handleCommit() {
+    // clear existing notifications
+    NotificationsApi.closeNotification(undefined, "committed");
+    NotificationsApi.closeNotification(undefined, "committing");
+    // display the committing notification
+    NotificationsApi.displayNotification(this.props.blueprint.name, "committing");
+    this.setNotifications();
+    // post blueprint (includes 'committed' notification)
+    Promise.all([BlueprintApi.handleCommitBlueprint(this.props.blueprint)])
+      .then(() => {
+        // then after blueprint is posted, reload blueprint details
+        // to get details that were updated during commit (i.e. version)
+        // and call create image
+        Promise.all([BlueprintApi.reloadBlueprintDetails(this.props.blueprint)])
+          .then(data => {
+            const blueprintToSet = Object.assign({}, this.props.blueprint, {
+              version: data[0].version
+            });
+            this.props.setBlueprint(blueprintToSet);
+            this.handleCreateImage();
+          })
+          .catch(e => console.log(`Error in reload blueprint details: ${e}`));
+      })
+      .catch(e => console.log(`Error in blueprint commit: ${e}`));
+  }
+
+  handleCreateImage() {
+    NotificationsApi.displayNotification(this.props.blueprint.name, "imageWaiting");
+    if (this.setNotifications) this.setNotifications();
+    if (this.handleStartCompose)
+      this.handleStartCompose(
+        this.props.blueprint.name,
+        this.state.imageType,
+        this.state.imageName,
+        this.state.uploadSettings
+      );
+    this.props.close();
+  }
+
+  handleStartCompose(blueprintName, composeType, imageName, uploadSettings) {
+    const upload = {
+      image_name: imageName,
+      settings: uploadSettings
+    };
+    this.props.startCompose(blueprintName, composeType, upload);
+  }
+
+  handleNextStep(activeStep, callback) {
```

Comment:
Can you rename this from callback to something more intuitive so that someone can understand what this callback is actually doing?

Reply:
Fixed